JavaScript/JavaScriptSDK/ajax/ajax.ts

```diff
@@ -1,528 +1,382 @@
-                                }
-                            }
-
-                            this.ajaxData.requestSentTime = dateTime.Now();
-                            this.ajaxData.loadingRequest = document.readyState === "loading";
+                xhr.ajaxData.requestSentTime = dateTime.Now();
+                xhr.ajaxData.loadingRequest = document.readyState === "loading";
 
-                            if (!this.ajaxData.onreadystatechangeCallbackAttached) {
+                if (!xhr.ajaxData.onreadystatechangeCallbackAttached) {
 
-                                // IE 8 and below does not support xmlh.addEventListener. This the last place for the browsers that does not support addEventListenener to intercept onreadystatechange
-                                var that = this;
-                                setTimeout(function () {
-                                    if (that.readyState === 4) {
+                    // IE 8 and below does not support xmlh.addEventListener. 
+                    // This is the last place for the browsers that does not support addEventListenener to instrument onreadystatechange
 
-                                        // ajax is cached, onreadystatechange didn't fire, but it is completed
-                                        commands.TryCatchTraceWrapper.call(that, "readyState(4)", collectResponseData);
-                                        onAjaxComplete.call(that);
-                                    }
-                                    else {
-                                        interceptOnReadyStateChange.call(that);
-                                    }
-                                }, 5);
+                    var ajaxMonitorInstance = this;
+                    setTimeout(function () {
+                        try {
+                            if (xhr.readyState === 4) {
+                                // ajax is cached, onreadystatechange didn't fire, but it is completed
+                                ajaxMonitorInstance.collectResponseData(xhr);
+                                ajaxMonitorInstance.onAjaxComplete(xhr)
                             }
-                        });
-                    }
-                }
-
-                function interceptAbort() {
-                    var originalAbort = XMLHttpRequest.prototype.abort;
-                    XMLHttpRequest.prototype.abort = function () {
-                        if (isMonitoredInstance.call(this)) {
-                            this.ajaxData.aborted = 1;
+                            else {
+                                ajaxMonitorInstance.instrumentOnReadyStateChange(xhr);
+                            }
+                        } catch (e) {
+                            _InternalLogging.throwInternalNonUserActionable(
+                                LoggingSeverity.CRITICAL,
+                                "Failed to instrument XMLHttpRequest.onreadystatechange"
```

Comment:
We have a protection against too many internal logs - https://github.com/Microsoft/ApplicationInsights-JS/blob/master/JavaScript/JavaScriptSDK/Logging.ts#L111

As for "something is very broken" - this is an assumption, perhaps valid, but still an assumption. Let's get errors first and then act on them.